homeassistant/components/hvac/__init__.py

```diff
@@ -0,0 +1,492 @@
+                "Received call to %s without attribute %s",
+                SERVICE_SET_SWING, ATTR_SWING_MODE)
+            return
+
+        for hvac in target_hvacs:
+            hvac.set_swing(swing_mode)
+
+            hvac.update_ha_state(True)
+
+    hass.services.register(
+        DOMAIN, SERVICE_SET_SWING, swing_set_service,
+        descriptions.get(SERVICE_SET_SWING))
+    return True
+
+
+class HvacDevice(Entity):
+    """Representation of a hvac."""
+
+    # pylint: disable=too-many-public-methods,no-self-use
+    @property
+    def state(self):
+        """Return the current state."""
+        return self.target_temperature or STATE_UNKNOWN
+
+    @property
+    def state_attributes(self):
+        """Return the optional state attributes."""
+        data = {
+            ATTR_CURRENT_TEMPERATURE:
+            self._convert_for_display(self.current_temperature),
+            ATTR_MIN_TEMP: self._convert_for_display(self.min_temp),
+            ATTR_MAX_TEMP: self._convert_for_display(self.max_temp),
+            ATTR_TEMPERATURE:
+            self._convert_for_display(self.target_temperature),
+            ATTR_HUMIDITY: self.current_humidity,
+            ATTR_MIN_HUMIDITY: self.min_humidity,
+            ATTR_MAX_HUMIDITY: self.max_humidity,
+            ATTR_FAN_LIST: self.fan_list,
+            ATTR_OPERATION_LIST: self.operation_list,
+            ATTR_SWING_LIST: self.swing_list,
+        }
+
+        current_operation = self.current_operation
+        if current_operation is not None:
+            data[ATTR_OPERATION] = current_operation
+
+        is_away = self.is_away_mode_on
+        if is_away is not None:
+            data[ATTR_AWAY_MODE] = STATE_ON if is_away else STATE_OFF
+
+        is_aux_heat = self.is_aux_heat_on
+        if is_aux_heat is not None:
+            data[ATTR_AUX_HEAT] = STATE_ON if is_aux_heat else STATE_OFF
+
+        current_fan_mode = self.current_fan_mode
+        if current_fan_mode is not None:
+            data[ATTR_FAN_MODE] = current_fan_mode
+
+        current_swing_mode = self.current_swing_mode
+        if current_swing_mode is not None:
+            data[ATTR_SWING_MODE] = current_swing_mode
+
+        return data
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        raise NotImplementedError
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return None
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        return None
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return None
+
+    @property
+    def operation_list(self):
+        """List of available operation modes."""
+        return None
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        raise NotImplementedError
```

Comment:
Good question. I guess they do. Because it is a combined fan/heat/thermostat controller.